src/main/java/org/codeforamerica/messaging/models/TemplateSet.java

```diff
@@ -0,0 +1,32 @@
+package org.codeforamerica.messaging.models;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import lombok.*;
+import org.codeforamerica.messaging.utils.RegexPatternStrings;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import java.time.OffsetDateTime;
+
+@Entity
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@ToString(onlyExplicitlyIncluded = true)
+public class TemplateSet {
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    private Long id;
+    @NotBlank
+    @Column(unique=true)
+    @Pattern(regexp = RegexPatternStrings.TEMPLATE_NAME_REGEX)
```

Comment:
I do not think it is useful to restrict what can go into a name. I do see that some clients could make it difficult for themselves by not choosing wise names. At this stage, it seems excessive to restrict it. We can add this later if necessary. Right now, I see value in having names like "Document#8" and "CDC's PHE Notice", etc.